### PR TITLE
ci(eslint): Upgrade to eslint v9

### PR DESCRIPTION
**Pull request by ryan953**
Upgrades eslint to v9 and also switches the config from `.eslintrc.js` to `eslint.config.js` format.

All the related eslint packages are updated as well to the latest, except for [eslint-plugin-react-hooks](https://github.com/facebook/react/issues/31687) which is having a regression right now.

I've had to add more files to the ignore list because they are not included inside the tsconfig.json file. This is something to revisit so we can have more coverage.

Relates to https://github.com/getsentry/frontend-tsc/issues/82
Relates to https://github.com/getsentry/frontend-tsc/issues/83

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -186,24 +189,24 @@
     "@testing-library/react": "16.0.0",
     "@testing-library/user-event": "14.5.2",
     "@types/node": "^22.9.1",
-    "@typescript-eslint/eslint-plugin": "^8.8.1",
-    "@typescript-eslint/parser": "^8.8.1",
```
Comment (ryan953) [inline package.json:190]: these two are replaced by `typescript-eslint`

File: static/app/components/globalDrawer/components.tsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ interface DrawerHeaderProps {
   hideCloseButton?: boolean;
 }
 
-export const DrawerHeader = forwardRef(function _DrawerHeader(
```
Comment (ryan953) [inline static/app/components/globalDrawer/components.tsx:73]: the lint rules didn't treat this like a react component, it has to start with a capitalized char

File: eslint.config.mjs [hunk 3]
```diff
@@ -710,11 +733,11 @@ const appRules = {
...
-  'no-lookahead-lookbehind-regexp/no-lookahead-lookbehind-regexp': [
-    'error',
-    'no-lookbehind',
-    'no-negative-lookbehind',
-  ],
+  // 'no-lookahead-lookbehind-regexp/no-lookahead-lookbehind-regexp': [
+  //   'error',
+  //   'no-lookbehind',
+  //   'no-negative-lookbehind',
+  // ],
```
Comment (ryan953) [inline eslint.config.mjs:740]: This was giving me grief. I'd rather ship without it than wait longer.

File: eslint.config.mjs [hunk 4]
```diff
@@ -737,132 +760,217 @@ const strictRules = {
...
 };
 
-const extendsList = [
-  'plugin:jest/recommended',
-  'plugin:jest-dom/recommended',
-  'plugin:import/typescript',
-];
-if (detectDeprecations) {
-  extendsList.push('plugin:deprecation/recommended');
-}
```
Comment (ryan953) [inline eslint.config.mjs:747]: This bit isn't ported over yet

File: eslint.config.mjs [hunk 5]
```diff
@@ -1,6 +1,30 @@
...
+ * This is your friend:
+ * `npx eslint --inspect-config`
+ */
+import * as emotion from '@emotion/eslint-plugin';
+import {fixupPluginRules} from '@eslint/compat';
+// import eslint from '@eslint/js';
+import importPlugin from 'eslint-plugin-import';
+import jest from 'eslint-plugin-jest';
+import jestDom from 'eslint-plugin-jest-dom';
+// import noLookaheadLookbehindRegexp from 'eslint-plugin-no-lookahead-lookbehind-regexp';
```
Comment (evanpurkhiser) [inline eslint.config.mjs:14]: Should we remove these comments?

Comment (scttcper) [inline eslint.config.mjs:14]: i'd go back to 0.1.0 to get this working

File: eslint.config.mjs [hunk 6]
```diff
@@ -737,132 +760,217 @@ const strictRules = {
...
+        // https://typescript-eslint.io/packages/parser/#project
+        project: './tsconfig.json',
 
-  settings: {
-    react: {
-      version: '18.2.0', // React version, can not `detect` because of getsentry
+        // https://typescript-eslint.io/packages/parser/#projectservice
+        // `projectService` is slower with our current tsconfig files.
+        // projectService: true,
+        // tsconfigRootDir: import.meta.dirname,
```
Comment (ryan953) [inline eslint.config.mjs:814]: `projectService` is recommended, but it's much slower. With `project` runtime is about 2m30s and with `projectService` we're at 3m

File: eslint.config.mjs (hunk 5 above)
Comment (ryan953) [reply eslint.config.mjs:14]: 0.1.0 isn't working either. I want to followup and fix the source if we have to in order to make it compatible with eslint v9 then re-enable this.

File: eslint.config.mjs [hunk 7]
```diff
@@ -737,132 +760,217 @@ const strictRules = {
...
+ *  - https://eslint.org/docs/latest/use/configure/configuration-files#specifying-files-and-ignores
+ *  - https://eslint.org/docs/latest/use/configure/configuration-files#cascading-configuration-objects
+ */
+export default typescript.config([
+  {
+    /**
+     * Main parser & linter options
+     * Rules are defined below and inherit these properties
+     * https://eslint.org/docs/latest/use/configure/configuration-files#configuration-objects
+     */
```
[the remark this answers is not in the document]
Comment (ryan953) [reply eslint.config.mjs:778]: the property is correctly documented by eslint, and idk if there's anywhere else too really see it. 

SCR-20241230-kuec

File: .github/workflows/frontend-lint-burndown.yml [hunk 8]
```diff
@@ -17,9 +17,9 @@ jobs:
     steps:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
 
-      - name: Install dependencies & inject eslint-plugin-deprecation
```
Comment (ryan953) [inline .github/workflows/frontend-lint-burndown.yml:20]: The rules inside this plugin are built-in to [typescript-eslint](https://typescript-eslint.io/)  now. Enabling it adds 1m of runtime and requires that `jsDocParsingMode: 'all'` be set. 

See https://typescript-eslint.io/packages/parser/#jsdocparsingmode

https://www.npmjs.com/package/eslint-plugin-deprecation

File: eslint.config.mjs (hunk 3 above)
Comment (ryan953) [reply eslint.config.mjs]: removed! we might not need this any longer anyway, so no need to fix it up.

File: eslint.config.mjs (hunk 4 above)
Comment (ryan953) [reply eslint.config.mjs:747]: it is now!

File: eslint.config.mjs (hunk 5 above)
Comment (ryan953) [reply eslint.config.mjs]: removed import comments

File: eslint.config.mjs (hunk 7 above)
[the remark this answers is not in the document]
Comment (ryan953) [reply eslint.config.mjs]: tweaked some of the comment blocks.

File: eslint.config.mjs [hunk 9]
```diff
@@ -737,132 +750,222 @@ const strictRules = {
...
+      'fixtures/profiles/embedded.js',
+      'jest.config.ts',
+      'api-docs/**/*',
+      'src/sentry/static/sentry/js/**/*',
+      'src/sentry/templates/sentry/**/*',
+      'stylelint.config.js',
+    ],
+  },
+  /**
+   * GLobal Rules
```
Comment (evanpurkhiser) [inline eslint.config.mjs:873]:
```suggestion
   * Global Rules
```

**Review by evanpurkhiser — APPROVED** [review]
let's slam it in

**Pull request merged**